--- a/server/async.c
+++ b/server/async.c
@@ -337,3 +337,13 @@ void async_wake_up( struct async_queue *queue, unsigned int status )
         if (status == STATUS_ALERTED) break;  /* only wake up the first one */
     }
 }
+
+/* update an async to correspond to new file object information */
+void async_update( struct async_queue *queue )
+{
+    if (!queue) return;
+
+    if (!queue->completion && queue->fd)
+        queue->completion = fd_get_completion( queue->fd, &queue->comp_key );
+}
+
