--- a/server/async.c
+++ b/server/async.c
@@ -265,7 +265,9 @@ void async_set_result( struct object *obj, unsigned int status, unsigned int tot
         if (async->timeout) remove_timeout_user( async->timeout );
         async->timeout = NULL;
         async->status = status;
-        if (async->queue->completion && async->data.cvalue)
+        if (status == STATUS_MORE_PROCESSING_REQUIRED)
+            async->status = STATUS_PENDING;
+        else if (async->queue->completion && async->data.cvalue)
             add_completion( async->queue->completion, async->queue->comp_key, async->data.cvalue, status, total );
         if (apc)
         {
