--- a/server/signal.c
+++ b/server/signal.c
@@ -90,7 +90,8 @@ static const struct fd_ops handler_fd_ops =
     NULL,                     /* ioctl */
     NULL,                     /* queue_async */
     NULL,                     /* reselect_async */
-    NULL                      /* cancel_async */
+    NULL,                     /* cancel_async */
+    NULL                      /* update_async */
 };
