--- a/server/named_pipe.c
+++ b/server/named_pipe.c
@@ -173,6 +173,7 @@ static const struct fd_ops pipe_server_fd_ops =
     default_fd_queue_async,       /* queue_async */
     default_fd_reselect_async,    /* reselect_async */
     default_fd_cancel_async,      /* cancel_async */
+    default_fd_update_async       /* update_async */
 };
 
 /* client end functions */
@@ -212,7 +213,8 @@ static const struct fd_ops pipe_client_fd_ops =
     default_fd_ioctl,             /* ioctl */
     default_fd_queue_async,       /* queue_async */
     default_fd_reselect_async,    /* reselect_async */
-    default_fd_cancel_async       /* cancel_async */
+    default_fd_cancel_async,      /* cancel_async */
+    default_fd_update_async       /* update_async */
 };
 
 static void named_pipe_device_dump( struct object *obj, int verbose );
@@ -256,7 +258,8 @@ static const struct fd_ops named_pipe_device_fd_ops =
     named_pipe_device_ioctl,          /* ioctl */
     default_fd_queue_async,           /* queue_async */
     default_fd_reselect_async,        /* reselect_async */
-    default_fd_cancel_async           /* cancel_async */
+    default_fd_cancel_async,          /* cancel_async */
+    default_fd_update_async           /* update_async */
 };
 
 static void named_pipe_dump( struct object *obj, int verbose )
