--- a/server/thread.c
+++ b/server/thread.c
@@ -157,7 +157,8 @@ static const struct fd_ops thread_fd_ops =
     NULL,                       /* ioctl */
     NULL,                       /* queue_async */
     NULL,                       /* reselect_async */
-    NULL                        /* cancel_async */
+    NULL,                       /* cancel_async */
+    NULL                        /* update_async */
 };
 
 static struct list thread_list = LIST_INIT(thread_list);
